daemon/config/config.go

```diff
@@ -392,13 +391,17 @@ func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Con
 	if err != nil {
 		return nil, err
 	}
+	// trim leading and trailing spaces
+	b = bytes.TrimSpace(b)
+	// accept empty config files, set content to {}
+	if len(b) == 0 {
+		b = []byte("{}")
+	}
 
 	var config Config
```

Comment:
I'm inclined to do the reverse here; not convert "empty" to `{}`, but check if it's empty, and if so, return early:

```suggestion
	var config Config

	b = bytes.TrimSpace(b)
	if len(b) == 0 {
		// empty config file
		return &config, nil
	}

```

Looks like all the code below should only be for loading the JSON file, and for validating if there's conflicts, so we wouldn't have to run that code afaics (yeah, this code and logic needs a rewrite, as it's quite messy)